### PR TITLE
format clouds.yaml for extension

### PR DESCRIPTION
**Pull request by clyang82**
Signed-off-by: clyang82 <clyang@cn.ibm.com>

This PR is for #62

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -89,11 +89,14 @@ Participation in the Kubernetes community is governed by the [Kubernetes Code of
    #### Quick notes on clouds.yaml
    ```shell
    $ cat clouds.yaml
-   userName: "your-name"
-   apiKey: "your-api-key"
+   clouds:
+    ibmcloud:
+      auth:
+        username: "test@test.com"
```
Comment (gyliu513) [inline README.md]: How about using:

```
apiUserName: "Your API Username"
AuthenticationKey: "Your API Authentication Key"
```

@jichenjc @xunpan WDYT?

Comment (gyliu513) [reply README.md]: The above proposal also align with the SL UI output.

Comment (clyang82) [reply README.md]: That makes sense. Let me modify to align with SL UI output

File: README.md [hunk 2]
```diff
@@ -89,11 +89,14 @@ Participation in the Kubernetes community is governed by the [Kubernetes Code of
...
+        password: "here'syourpassword"
    ```
 
-   You can get `userName` and `apiKey` from https://control.softlayer.com/ .
+   You can get `username` and `password` from https://control.softlayer.com/ .
```
Comment (jichenjc) [inline README.md]: we won't get password, instead , we get apiKey..

File: pkg/cloud/types.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+// IbmcloudConfig holds ibm cloud provider required configurations
+type IbmcloudConfig struct {
+	Auth AuthConfig `yaml:"auth,omitempty"`
+}
+
+// AuthConfig is mounted into controller pod for clouds authentication
+type AuthConfig struct {
+	UserName string `yaml:"username,omitempty"`
+	Password string `yaml:"password,omitempty"`
```
Comment (jichenjc) [inline pkg/cloud/types.go]: again, please avoid password usage... public cloud should use api key to interact..

**Comment by xunpan** [comment]
@clyang82  Please squash commit next time if just fix previous commit issue in 1 PR. It should mess up log history.

**Pull request merged**